package.json

```
@@ -12,12 +12,16 @@
     "pretest": "lerna run pretest",
     "test": "node scripts/test.js",
     "posttest": "lerna run posttest",
-    "test:watch": "node scripts/test.js --watch",
-    "lint": "tsc -b && yarn prettier:check && eslint --ext .js,.jsx,.ts,.tsx,.graphql .",
-    "lint:fix": "eslint --ext .js,.jsx,.ts,.tsx,.graphql . --fix; prettier --ignore-path .eslintignore --write '**/*.{js,jsx,ts,tsx,graphql,md}'",
-    "prettier:check": "prettier --ignore-path .eslintignore --check '**/*.{js,jsx,ts,tsx,graphql,md}'",
-    "depcheck": "lerna exec --stream 'yarn depcheck --ignores=\"@app/config,@app/client,tslib,webpack,babel-plugin-import,source-map-support,@graphql-codegen/*,graphql-toolkit,net,tls,dayjs\" --ignore-dirs=\".next\"'",
-    "dev": "yarn && lerna run codegen --stream && tsc -b && concurrently --kill-others --names 'TSC,WATCH,RUN,TEST' --prefix '({name})' --prefix-colors 'yellow.bold,yellow.bold,cyan.bold,greenBright.bold' 'tsc -b --watch --preserveWatchOutput --assumeChangesOnlyAffectDirectDependencies' 'lerna run --parallel watch' 'lerna run --parallel dev' 'sleep 10; yarn test:watch'",
+    "test:watch": "node scripts/test.js --watch --delay 10",
+    "lint": "tsc -b && yarn prettier:check && yarn eslint:check",
+    "lint:fix": "run-script-os",
+    "lint:fix:windows": "yarn eslint:check --fix & yarn prettier:fix",
+    "lint:fix:default": "yarn eslint:check --fix; yarn prettier:fix",
+    "eslint:check": "eslint --ext .js,.jsx,.ts,.tsx,.graphql .",
+    "prettier:check": "prettier --ignore-path .eslintignore --check \"**/*.{js,jsx,ts,tsx,graphql,md}\"",
+    "prettier:fix": "prettier --ignore-path .eslintignore --write \"**/*.{js,jsx,ts,tsx,graphql,md}\"",
```

I think we can refactor this so that we extract the path spec parts. This also allows users to run `yarn eslint path/to/directory/` and have it do what they want (respecting all the extensions). `yarn prettier path/to/file.ts` will also work, which is why I've named the script that includes all the paths in the project `prettier:all`.

```suggestion
    "lint": "tsc -b && yarn prettier:all --check && yarn eslint .",
    "lint:fix": "run-script-os",
    "lint:fix:windows": "yarn eslint --fix . & yarn prettier:all --write",
    "lint:fix:default": "yarn eslint --fix .; yarn prettier:all --write",
    "eslint": "eslint --ext .js,.jsx,.ts,.tsx,.graphql",
    "prettier:all": "prettier --ignore-path .eslintignore \"**/*.{js,jsx,ts,tsx,graphql,md}\"",
```

---

Despite what I said previously I think I'm okay with changing to simply only run prettier if ESLint passes:

```
  "lint:fix": "yarn eslint --fix . && yarn prettier:all --write"
```

if this removes the need for `run-script-os`. It's a bit redundant to have three lines just so that we can use `&` / `;`. Sorry for the back-and-forth on this.